File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    private List<int> valList;
    public List<int> RightSideView(TreeNode root) {
        valList = new();

        AddRightSideView(root, 0);

        return valList;

    }

    private void AddRightSideView(TreeNode curr, int depth){
        if (curr == null)
            return;
        
        if (valList.Count <= depth)
        {
            valList.Add(curr.val);
        }

        depth++;
        AddRightSideView(curr.right, depth);
        AddRightSideView(curr.left, depth);
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    count = 0
    def goodNodes(self, root: TreeNode) -> int:
        count = 0
        self.checkNode(root, root.val)

        return self.count

    def checkNode(self, node: Optional[TreeNode], pathMax: int):
        if node is None:
            return

        if node.val >= pathMax:
            self.count += 1
            pathMax = max(node.val, pathMax)
        
        self.checkNode(node.left, pathMax)
        self.checkNode(node.right, pathMax)
        